test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -85,6 +90,7 @@
     private static final String BOOKKEEPER_IMAGE_NAME = System.getProperty("bookkeeperImageName", "bookkeeper");
     private static final String TIER2_NFS = "nfs";
     private static final String TIER2_TYPE = System.getProperty("tier2Type", TIER2_NFS);
+    private static final boolean IS_OPERATOR_VERSION_ABOVE_040 = isOperatorVersionAbove040();
```

Comment:
Again, the change is only focused on the incompatibility before/post `0.4`. So if there is a new change in the operator that requires additional management here (e.g., between operator versions `0.9` and `1.X`), then we will need to refactor this class and then start working with "version ranges". I don't see the problem about starting to work in that way right from the beginning.

Reply:
Could you please explain why this change cannot be introduced when it is needed?

I don't want to hold off this PR, for some future requirements that may not be needed for a very long time to come and don't look very compelling to me at this point.